### ISSUE TITLE
Add the ability to embed the application

### ISSUE THREAD

**Issue by uncle-yura** [mbuesch/awlsim#28]
Now I am making the Linuxcnc GUI on pyqt and want to embed awlsim in one of the tabs.
But starting with this code leads to a number of errors or freezes:
```
from awlsim.gui.mainwindow import *
class AwlsimWidget(QWidget):
    def __init__(self, parent=None, path=None):
        QWidget.__init__(self, parent)
        try:
            mainwnd = MainWindow.start()
            layout = QHBoxLayout()
            layout.addWidget(mainwnd)
            self.setLayout(layout)
        except Exception as e:
            printDebug(_("Failed to import awlsim, {}",e))
```
Can you add a parameter to start awlism process in the window?
Like this:
```
self.process = QtCore.QProcess(self)
self.process.start('xterm',['-into', str(int(self.winId())), '-geometry', "600x60+0+400"])
```

**Comment by mbuesch** [comment]
Thanks for your suggestion.

I agree that it would be a good thing to make Awlsim fully embeddable.
I think it already is partially embeddable. For instance the editor widgets should be usable on their own. Or the ProjectTree.
Also the MainWidget should be usable on its own. However it used to be tightly coupled to the MainWindow. I'm not sure how much of that coupling is still left. But that could certainly be fixed.

But I'm not so sure about embedding the MainWindow for a couple of reasons. First being: Do we really want that? The main window is already pretty crowded and things feel misplaced (e.g. the CPU state area). Especially on small screens. That's not going to get better, if we embed it. The second doubt would be: I'm not sure if that's even technically easily possible. The MainWindow is based on QMainWindow. That class is by definition _the_ application main window. And it does some things that probably make it hard to embed it somewhere.
I have no idea what X11 tricks could be played to embed the QMainWindow into another application.

Last but not least: The internal APIs in the awlsim.gui module are not stable. They are going to change with every release. They are not public API and that's not going to change for most of them. So updating Awlsim would be a major effort for the user application developer. I could imagine some sort of public and stable API for the GUI. but that would have to be carefully designed.

**Comment by uncle-yura** [comment]
Yes, you're right, there is very little space left on the small screen. 
I think it would be nice to make a compact version. 
In addition, free windows create confusion. 
I'll think about what can be done about it.
